### ISSUE TITLE
Should a Gt PreparedStatementException take-on the code from the exception it wraps?

### ISSUE THREAD

**Issue by j4m3s** [PhpGt/Database#48]
Either that, or maybe implement some more specific exceptions (such as DuplicateKeyException).  Currently, the PreparedStatementException has a code of 0 - you have to call getPrevious to access the underlying PDOException and access its error code.  

If we don't either inherit the code or throw some more specific exceptions it would be better to simply drop the wrapper exception and let the PDOException bubble up.

**Comment by g105b** [comment]
Agreed, exceptions are not as intuitive as they should be so far. Passing the code is a good idea, haven't given it much thought yet.
